daft/series.py

```diff
@@ -51,19 +64,37 @@ def from_arrow(array: pa.Array | pa.ChunkedArray, name: str = "arrow_series") ->
             return Series.from_pylist(array.to_pylist(), name=name, pyobj="force")
         elif isinstance(array, pa.Array):
             array = ensure_array(array)
-            pys = PySeries.from_arrow(name, array)
-            return Series._from_pyseries(pys)
+            if _RAY_DATA_EXTENSIONS_AVAILABLE and isinstance(array.type, ArrowTensorType):
+                storage_series = Series.from_arrow(array.storage, name=name)
+                series = storage_series.cast(
+                    DataType.fixed_size_list(
+                        "item", DataType.from_arrow_type(array.type.scalar_type), int(np.prod(array.type.shape))
+                    )
+                )
+                return series.cast(DataType.from_arrow_type(array.type))
+            elif _RAY_DATA_EXTENSIONS_AVAILABLE and isinstance(array.type, ArrowVariableShapedTensorType):
+                return Series.from_numpy(array.to_numpy(zero_copy_only=False), name=name)
+            elif isinstance(array.type, getattr(pa, "FixedShapeTensorType", ())):
+                series = Series.from_arrow(array.storage, name=name)
+                return series.cast(DataType.from_arrow_type(array.type))
+            else:
+                pys = PySeries.from_arrow(name, array)
+                return Series._from_pyseries(pys)
         elif isinstance(array, pa.ChunkedArray):
             array = ensure_chunked_array(array)
             arr_type = array.type
-            if isinstance(arr_type, pa.BaseExtensionType):
+            if _RAY_DATA_EXTENSIONS_AVAILABLE and isinstance(
+                arr_type, (ArrowTensorType, ArrowVariableShapedTensorType)
+            ):
+                from ray.air.util.transform_pyarrow import _concatenate_extension_column
+
+                combined_array = _concatenate_extension_column(array)
+            elif isinstance(arr_type, pa.BaseExtensionType):
                 combined_storage_array = array.cast(arr_type.storage_type).combine_chunks()
                 combined_array = arr_type.wrap_array(combined_storage_array)
```

Comment:
Ah, so I initially did this as an intentional deviation, but now that I think about it, it's probably not needed.

I deviated here since our generic extension type chunk concatenation logic won't properly handle the case when chunks have incompatible fixed-shape types (e.g. concatenating `FixedShapeTensor(int64, (2, 2))` and `FixedShapeTensor(int64, (3, 3))` arrays should result in a variable-shape `Tensor(int64)` array), and this is primarily what `_concatenate_extension_column()` from Ray handles. But we don't actually need this case since chunks must all have the exact same type.

I'll remove this case!